### PR TITLE
Use date library as a library

### LINKED ISSUE #176: The date library should be treated as a library
Related to issue #157.

Seeing as the date library is quite a bit of code that is continually being updated and that it has been submitted for inclusion into the C++ standard, we should probably include it via conan instead of as source code in the repository.

### PR DESCRIPTION
**Pull request by mattclarke**
### Description of work

Removes the Date source from the code base and use conan instead

### Issue

Closes #176

### Acceptance Criteria

Build still works. No code changes were made

### Unit Tests

N/A

### Other

N/A

---

## Code Review (To be filled in by the reviewer only)

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel (does someone want to automate this?).

### CODE REVIEW

**Comment by SkyToGround** [comment]
Looks good to me. I will approve it when the build has completed successfully.

**Comment by mattclarke** [comment]
@rerpha and @mchorazak did the work. I just sorted out the conan-date bit

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -143,11 +144,6 @@ set(kafka_to_nexus_INC
   MessageTimestamp.h
   TopicReader.h
   uri.h
-  date/date.h
```
Comment (SkyToGround) [inline src/CMakeLists.txt]: :+1:

**Comment by dominikwerder** [comment]
Any idea why `date` requires c++17?

I don't specify any language standard in my conan profile, don't know if that would even be possible.

```
date/4b46deb@ess-dmsc/stable: Copying sources to build folder
date/4b46deb@ess-dmsc/stable: Generator cmake created conanbuildinfo.cmake
date/4b46deb@ess-dmsc/stable: Calling build()
-- The C compiler identification is GNU 6.3.1
-- The CXX compiler identification is GNU 6.3.1
-- Check for working C compiler: /opt/rh/devtoolset-6/root/usr/bin/cc
-- Check for working C compiler: /opt/rh/devtoolset-6/root/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/rh/devtoolset-6/root/usr/bin/c++
-- Check for working CXX compiler: /opt/rh/devtoolset-6/root/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
# date: USE_SYSTEM_TZ_DB True
# date: USE_TZ_DB_IN_DOT False
# date: BUILD_SHARED_LIBS ON
# date: ENABLE_DATE_TESTING False
# date: DISABLE_STRING_VIEW True
-- Configuring done
CMake Error at CMakeLists.txt:48 (add_library):
  CXX_STANDARD is set to invalid value '17'

```

**Pull request merged**